bucket_map/src/bucket.rs

```diff
@@ -866,162 +1138,169 @@ mod tests {
+            for max_search in [2usize, 3] {
+                for v in 10..12u64 {
+                    for len in 1..(max_search + 1) {
+                        let raw = (0..len)
+                            .map(|l| {
+                                let k = Pubkey::from([l as u8; 32]);
+                                (k, v + (l as u64))
+                            })
+                            .collect::<Vec<_>>();
+                        let mut hashed = Bucket::index_entries(&raw, random);
+                        let common_ix = 2; // both are put at same ix
+                        hashed.iter_mut().for_each(|v| {
+                            v.0 = common_ix;
+                        });
+                        let hashed_raw = hashed.clone();
+
+                        let mut index = create_test_index(Some(max_search as u8));
+
+                        let mut duplicates = Vec::default();
+                        let mut entries_created = 0;
+                        let result = Bucket::<u64>::batch_insert_non_duplicates_internal(
+                            &mut index,
+                            &Vec::default(),
+                            &raw,
+                            &mut hashed,
+                            &mut entries_created,
+                            &mut duplicates,
+                            try_to_reuse_disk_data,
+                        );
+
+                        assert_eq!(
+                            hashed.len(),
+                            if len > max_search { 1 } else { 0 },
+                            "len: {len}"
+                        );
+                        (0..len).for_each(|i| {
+                            assert!(if len > max_search {
+                                result.is_err()
+                            } else {
+                                result.is_ok()
+                            });
+                            let raw2 = hashed_raw[i];
+                            if i == 0 && len > max_search {
+                                // max search was exceeded and the first entry was unable to be inserted, so it remained in `hashed`
+                                assert_eq!(hashed[0], hashed_raw[0]);
+                            } else {
+                                // we insert in reverse order when ix values are equal, so we expect to find item[1] in item[1]'s expected ix and item[0] will be 1 search distance away from expected ix
+                                let search_required = (len - i - 1) as u64;
+                                let elem = IndexEntryPlaceInBucket::new(
+                                    (raw2.0 + search_required) % index.capacity(),
+                                );
+                                let (value, ref_count) = elem.read_value(&index, &data_buckets);
+                                assert_eq!(ref_count, 1);
+                                assert_eq!(value, &[raw[hashed_raw[i].1].1]);
```

Comment:
It seems that the test filed at this line.

Reply:
fixed it